### PR TITLE
thorvg: Fix MinGW build by adding explicit `<cstring>` include

### LINKED ISSUE #59044: Cross compiling windows export template from arch linux errors in tvgSvgCSSStyle
### Godot version

4.0dev

### System information

Arch Linux

### Issue description

Cross compiling a windows build from arch linux results in this error

```
[ 42%] Compiling thirdparty/thorvg/src/loaders/svg/tvgSvgCssStyle.cpp ...
[ 43%] Compiling thirdparty/thorvg/src/loaders/tvg/tvgTvgLoader.cpp ...
[ 43%] Compiling thirdparty/thorvg/src/loaders/jpg/tvgJpgLoader.cpp ...
[ 43%] Compiling thirdparty/thorvg/src/loaders/jpg/tvgJpgd.cpp ...
[ 43%] Compiling thirdparty/thorvg/src/loaders/external_png/tvgPngLoader.cpp ...
[ 43%] Compiling thirdparty/thorvg/src/lib/sw_engine/tvgSwFill.cpp ...
[ 43%] Compiling thirdparty/thorvg/src/lib/sw_engine/tvgSwImage.cpp ...
[ 43%] Compiling thirdparty/thorvg/src/lib/sw_engine/tvgSwMath.cpp ...
[ 43%] Compiling thirdparty/thorvg/src/lib/sw_engine/tvgSwMemPool.cpp ...
thirdparty/thorvg/src/loaders/svg/tvgSvgCssStyle.cpp: In function 'void _copyStyle(SvgStyleProperty*, const SvgStyleProperty*)':
thirdparty/thorvg/src/loaders/svg/tvgSvgCssStyle.cpp:43:56: error: 'strdup' was not declared in this scope
   43 |         if (from->fill.paint.url) to->fill.paint.url = strdup(from->fill.paint.url);
      |                                                        ^~~~~~
thirdparty/thorvg/src/loaders/svg/tvgSvgCssStyle.cpp:62:60: error: 'strdup' was not declared in this scope
   62 |         if (from->stroke.paint.url) to->stroke.paint.url = strdup(from->stroke.paint.url);
      |                                                            ^~~~~~
thirdparty/thorvg/src/loaders/svg/tvgSvgCssStyle.cpp: In function 'void cssCopyStyleAttr(SvgNode*, const SvgNode*)':
thirdparty/thorvg/src/loaders/svg/tvgSvgCssStyle.cpp:123:62: error: 'strdup' was not declared in this scope
  123 |     if (from->style->clipPath.url) to->style->clipPath.url = strdup(from->style->clipPath.url);
      |                                                              ^~~~~~
thirdparty/thorvg/src/loaders/svg/tvgSvgCssStyle.cpp:124:54: error: 'strdup' was not declared in this scope
  124 |     if (from->style->mask.url) to->style->mask.url = strdup(from->style->mask.url);
      |                                                      ^~~~~~
thirdparty/thorvg/src/loaders/svg/tvgSvgCssStyle.cpp: In function 'SvgNode* cssFindStyleNode(const SvgNode*, const char*, SvgNodeType)':
thirdparty/thorvg/src/loaders/svg/tvgSvgCssStyle.cpp:135:73: error: 'strcmp' was not declared in this scope
  135 |             if ((!title && !(*child)->id) || (title && (*child)->id && !strcmp((*child)->id, title))) return (*child);
      |                                                                         ^~~~~~
thirdparty/thorvg/src/loaders/svg/tvgSvgCssStyle.cpp:24:1: note: 'strcmp' is defined in header '<cstring>'; did you forget to '#include <cstring>'?
   23 | #include "tvgSvgCssStyle.h"
  +++ |+#include <cstring>
   24 |
thirdparty/thorvg/src/loaders/svg/tvgSvgCssStyle.cpp: In function 'SvgNode* cssFindStyleNode(const SvgNode*, const char*)':
thirdparty/thorvg/src/loaders/svg/tvgSvgCssStyle.cpp:149:44: error: 'strcmp' was not declared in this scope
  149 |             if ((title && (*child)->id && !strcmp((*child)->id, title))) return (*child);
      |                                            ^~~~~~
thirdparty/thorvg/src/loaders/svg/tvgSvgCssStyle.cpp:149:44: note: 'strcmp' is defined in header '<cstring>'; did you forget to '#include <cstring>'?
[ 43%] Compiling modules/text_server_adv/script_iterator.cpp ...
scons: *** [thirdparty/thorvg/src/loaders/svg/tvgSvgCssStyle.windows.double.opt.64.o] Error 1
scons: building terminated because of errors.
[Time elapsed: 00:00:11.156]
```

### Steps to reproduce

scons -j10 platform=windows tools=no target=release bits=64 float=64 

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by akien-mga**
Fixes #59044.

I didn't bother making a patch file / identifying the downstream change as it's a trivial fix which is likely going to be merged upstream shortly, and part of the next version (PR'ed upstream with https://github.com/Samsung/thorvg/pull/1206).

### CODE REVIEW

File: platform/windows/display_server_windows.cpp [hunk 1]
```diff
@@ -3596,12 +3596,11 @@ DisplayServerWindows::DisplayServerWindows(const String &p_rendering_driver, Win
 			return;
 		}
 
-		//gl_manager->set_use_vsync(current_videomode.use_vsync);
 		RasterizerGLES3::make_current();
 	}
 #endif
 
-	HHOOK mouse_monitor = SetWindowsHookEx(WH_MOUSE, ::MouseProc, nullptr, GetCurrentThreadId());
+	SetWindowsHookEx(WH_MOUSE, ::MouseProc, nullptr, GetCurrentThreadId());
```
Comment (akien-mga) [inline platform/windows/display_server_windows.cpp:3603]: @bruvzg Let me know if there should some error handling or similar instead.

Comment (bruvzg) [inline platform/windows/display_server_windows.cpp:3603]:
```suggestion
	mouse_monitor = SetWindowsHookEx(WH_MOUSE, ::MouseProc, nullptr, GetCurrentThreadId());
```

`mouse_monitor` is supposed to be used to deinit it in the destructor.

Comment (akien-mga) [reply platform/windows/display_server_windows.cpp]: Ah I see, thanks. It's no as benign as I thought then, I'll move it to a dedicated PR.

Comment (akien-mga) [reply platform/windows/display_server_windows.cpp]: #59199

**Pull request merged**